### ISSUE TITLE
[Question]: Visual comparisons Feature Parity 

### ISSUE THREAD

**Issue by infinitejest69** [microsoft/playwright-dotnet#1854]
### Your question

Hi Just wanted to check what the approach will be in terms of feature parity with the JavaScript version of playwright.

I was looking for some basic visual testing and noticed the JS version offers this via Pixelmatch 
[test-snapshots](https://playwright.dev/docs/test-snapshots)

Is the plan that the dotnet version will have feature parity with JS version or do we need to look at other implementations to try achieve the same feature of the Java script version.

**Comment by pavelfeldman** [comment]
Playwright Library has feature parity across languages. Visual regression testing is a part of the @playwright/test, a test runner for Playwright on Node.js, so it is missing on .Net.

It 1.18, you will likely see https://playwright.dev/docs/next/api/class-locatorassertions/ for .Net. And only after that, (1.19?), we are likely to add visual assertions into that set.

**Comment by infinitejest69** [comment]
Thanks for the reply.
is there a backlog or some roadmap where we can upvote features we would like to see ported before others or is features done on a case by case basis

**Comment by Skintkingle** [comment]
I would echo the OPs question. We're looking to adopt Playwright .Net into our testing pipeline and we were very excited to see Visual comparisons in the docs, and then sad when we realised it wasn't in the .Net flavour. Is there a feature upvoting site we can contribute to to help prioritise this higher?

**Comment by infinitejest69** [comment]
Just going to give this a bump now 1.18 is out

are we likely to see this feature enabled within playwright or does anyone know good open source alternatives that will play nicely with playwright ?

**Comment by infinitejest69** [comment]
Updating this for future travels who get their question marked as answered with this post
Thanks to @vmachacek 

You can use this to solve some of the issue with a few small lines of code
[Verify.HeadlessBrowsers](https://github.com/VerifyTests/Verify.HeadlessBrowsers)
and this blog that show how easy it is to use
[snapshot-testing-with-verify](https://www.danclarke.com/snapshot-testing-with-verify)

**Comment by ukaszef** [comment]
hey, any updates on the topic? i'm migrating my selenium-based framework to playwright in .net, visual testing is one of the important points for the nearest future - would love to be able to use playwright for this instead of external tooling. really sad to see it in js/ts and not in .net :(

**Comment by dotnetmaniac** [comment]
Hello Playwright team! 
Can we hope for this feature anytime in the near future?
We have been using Playwright for some time now, and we have come across this requirement and looks like we are in trouble as it's not supported in c#.

We would really appreciate any response/suggestions.

**Comment by zandraswo** [comment]
It is a pity this is not supported in dotnet version. It would provide a huge benefit in a lot of the tests. I wanted to migrate to Playwright and this is a huge dealbreaker. Hope this gets implemented soon.

**Comment by ddweber** [comment]
Any updates on this?
This feature is the missing piece that would allow us to migrate from tools like Ranorex and others to the easy and awesome playwright.

**Comment by Awwm12** [comment]
Also looking for an update on this as we're starting a new project soon that would really benefit from this. 
 Would really like to use Playwright with C#

**Comment by auslavs** [comment]
I remembered this issue from late 2021 when I wrote my first set of playwright tests.
It would still be nice to have this feature.

**Comment by MattHorrocksNetwealth** [comment]
I was just looking for this exact functionality in the docs. I've been doing some searching in the repo, and it looks like `ToMatchSnapshot` is used internally, or there's at least a PR to rename existing functionality to `ToMatchSnapshot`?

https://github.com/microsoft/playwright-dotnet/pull/2928

Can this be made public?

**Comment by mxschmitt** [comment]
The [ToMatchSnapshot](https://github.com/microsoft/playwright-dotnet/blob/0dc3195929f43c7fdbfc9c53ed60e74a9166b3a9/src/Playwright.Tests/BaseTests/PlaywrightAssert.cs#L59-L105) method inside this codebase is by far not as powerful as the one upstream in Node.js is. We don't feel confident making it public as of today. I recommend copying it into your own code base for now until we have figured out e.g. asset handling etc.

**Comment by kudlata** [comment]
Hi, 
I'm really waiting or this feature too. I hope it will be appear very soon :)
